zerver/views/messages.py

```diff
@@ -755,10 +753,8 @@ def same_realm_zephyr_user(user_profile, email):
     except ValidationError:
         return False
 
-    domain = email_to_domain(email)
-
     return user_profile.realm.is_zephyr_mirror_realm and \
-        RealmAlias.objects.filter(realm=user_profile.realm, domain=domain).exists()
+        get_realm_by_email_domain(email) == user_profile.realm
 
```

Comment:
@rishig We should not be querying the `RealmAlias` table directly since now there is no direct mapping between realm alias and realm, right?